### PR TITLE
MueLu: add documentation to InterfaceAggregationFactory

### PR DESCRIPTION
**Pull request by mayrmt**
@trilinos/muelu @kliegeois 

## Motivation

I'm working on using the `InterfaceAggregationFactory` in an application code. This PR is supposed to improve documentation of this (and associated) factories to make it easier for future users.

## Related Issues

* Follows PR #6931 

## Stakeholder Feedback

The `InterfaceAggregationFactory` has just been added recently. I'm not aware of its use in application codes so far other than by its author @kliegeois.

## Testing

This just adds documentation, so I rebuilt and double-checked Doxygen locally.

## Additional Information

@kliegeois Please have a look, if I described everything correctly. I'm happy to adapt it, if I got something wrong. Thanks in advance!

### CODE REVIEW

File: packages/muelu/src/Misc/MueLu_InterfaceAggregationFactory_decl.hpp [hunk 1]
```diff
@@ -56,6 +56,51 @@ namespace MueLu
 /*!
     @class InterfaceAggregationFactory class.
     @brief Factory for building aggregates for Lagrange multipliers.
+
+    ## Context, assumptions, and use cases ##
+
+    This factory is intended to be used for saddle-point systems of surface-coupled problems,
+    where constraints are enforced with Lagrange multipliers.
+    In addition to the primal unknowns, Lagrange multipliers are considered as dual unknowns.
+    The presenence of Lagrange mulitpliers make this a prima/dual problem
```
Comment (kliegeois) [inline packages/muelu/src/Misc/MueLu_InterfaceAggregationFactory_decl.hpp:65]: > The presenence of Lagrange mulitpliers make this a prima/dual problem

Should be replaced by:
The presence of Lagrange multipliers make this a primal/dual problem.

File: packages/muelu/src/Misc/MueLu_InterfaceAggregationFactory_decl.hpp [hunk 2]
```diff
@@ -56,6 +56,51 @@ namespace MueLu
...
+    A                       | Factory | null |   | * | * | Generating factory of the matrix A
+    Aggregates              | Factory | null |   | * | * | Generating factory of the aggregates (of type "Aggregates" produced, e.g., by the UncoupledAggregationFactory)
+    DualNodeID2PrimalNodeID | Factory | null |   | * | * | Generating factory of the fine dual-to-primal node mapping
+
+    The * in the @c master.xml column denotes that the parameter is defined in the @c master.xml file.<br>
+    The * in the @c validated column means that the parameter is declared in the list of valid input parameters (see InterfaceAggregationFactory::GetValidParameters).<br>
+    The * in the @c requested column states that the data is requested as input with all dependencies (see InterfaceAggregationFactory::DeclareInput).
+
+    The mapping of dual-to-primal node IDs, \c DualNodeID2PrimalNodeID, is of data type \c std::map<LocalOrdinal,LocalOrdinal>.
+    The 'key' refers to the global ID of the dual node, while the 'value' represents the global ID of its primal counterpart.
```
Comment (kliegeois) [inline packages/muelu/src/Misc/MueLu_InterfaceAggregationFactory_decl.hpp:94]: The IDs are actually stored as local ID and not global ID.
Therefore, the line:

> The 'key' refers to the global ID of the dual node, while the 'value' represents the global ID of its primal counterpart.

Should be replaced by:

The 'key' refers to the local ID of the dual node, while the 'value' represents the local ID of its primal counterpart.

We can argue on the use of a `std::map` and consider a `std::vector`. However, Moertel (the Trilinos package used to compute the Mortar matrices) relies on a map currently.

File: packages/muelu/src/Misc/MueLu_InterfaceAggregationFactory_def.hpp [hunk 3]
```diff
@@ -71,6 +71,10 @@ RCP<const ParameterList> InterfaceAggregationFactory<Scalar, LocalOrdinal, Globa
     DualNodeID2PrimalNodeID represents the mapping between the Dual Node IDs to the Primal Node IDs.
     This map makes one assumption: the number of dof per dual node is a constant and has to be set in 
     the parameter list as "number of DOFs per dual node".
+
+    Layout of the std::map<key, value>:
+    - key: dual node ID
```
Comment (kliegeois) [inline packages/muelu/src/Misc/MueLu_InterfaceAggregationFactory_def.hpp:76]: We can potentially stress one more time that it is a local ID.

File: packages/muelu/src/Misc/MueLu_InterfaceAggregationFactory_decl.hpp (hunk 2 above)
Comment (mayrmt) [reply packages/muelu/src/Misc/MueLu_InterfaceAggregationFactory_decl.hpp:94]: I'm fine with `std::map` if this enhances compatibility wit `Moertel`.

File: packages/muelu/src/Misc/MueLu_InterfaceAggregationFactory_def.hpp (hunk 3 above)
Comment (mayrmt) [reply packages/muelu/src/Misc/MueLu_InterfaceAggregationFactory_def.hpp:76]: Sure!

**Comment by mayrmt** [comment]
@kliegeois Thanks for you comments and fixes! I updated the branch. Please have another look.

**Comment by mayrmt** [comment]
@kliegeois Since the dual-to-primal node mapping uses local IDs, shall I rename the parameter list entry `"DualNodeID2PrimalNodeID"` to `"DualNodeLID2PrimalNodeLID"` to reflect the local ID character? I would address this along with #7164, if ok with you.

**Pull request merged**